docs/core/howto/ssl.rst

```diff
@@ -204,10 +199,40 @@ For example,
     options = CertificateOptions(..., dhParameters=dhParams)
 
 Another part of the TLS protocol which ``CertificateOptions`` can control is the version of the TLS or SSL protocol used.
-This is often called the context's "method".
+There are two ways to configure it, either by configuring a list of protocols, or by the legacy way of setting the internal context's "method".
+By default, Twisted will configure it to use TLSv1.0 or later and disable the insecure SSLv3 protocol, but this manual control over protocols can be helpful if you need to support legacy SSLv3 systems, or you wish to restrict it down to just the strongest of the TLS versions.
+
+You can give ``CertificateOptions`` a range of TLS versions you wish it to negotiate with the ``tlsProtocols`` argument in the initializer:
+
+.. code-block:: python
+
+    from twisted.internet.ssl import (CertificateOptions, TLSVersion,
+                                      TLSVersionRange)
+    options = CertificateOptions(
+        ...,
+        tlsProtocols=TLSVersionRange(TLSVersion.TLSv1_0,
+                                     TLSVersion.TLSv1_2))
+
```

Comment:
It sounds to me like we should explore the potential use-cases a bit here.

## 0. Don't Mess With Anything

Worth repeating that this is what _almost everyone_ should do; the customization API here is for edge cases.

## 1. Decrease minimum TLS version to be lower than that supported by Twisted by default in order to support older clients.

If you run a non-security-critical service and you want to consciously opt in to a bad, old version of TLS that isn't _so_ bad it's disabled yet, but has widespread deployment in your expected client base, 

## 2. Increase minimum TLS version to be higher than that required by Twisted by default in order to be more secure when you control both ends of the wire.

Twisted makes certain assumptions about your client base; if your clients are known, limiting to the most recent / best version of TLS is better than accepting potentially old/bad versions.

## 3. Limit maximum TLS version for security testing purposes (to emulate older clients).

If one is writing a testing tool like https://www.ssllabs.com/ssltest/, one needs a way to clamp the maximum TLS version that we will negotiate, mainly in order to ensure that our peer isn't doing something dumb.  This is _extremely esoteric_ and ideally would be hard to do by accident.

## 4. Skip an intermediate TLS version in order to hot-fix a future vulnerability

This is the one that @Lukasa's gut is telling us doesn't exist.

Keep in mind that this may be an _implementation_ vulnerability and not just a spec vulnerability though, which seems more plausible to me.  Especially if we were to (say) rapidly cycle from TLS 1.3 1.4 1.5, and 1.4 doesn't see a lot of deployment, it does make sense to me that there might be a "surprise" vulnerability in OpenSSL in the implementation of 1.4.

However, even if this case were to exist, the "right" fix is to get security updates for your TLS library or for Twisted, not to hack around it in your app.  So this strikes me as interesting mainly for completeness rather than "real" usage; the utility in doing it now would be to make it easy to do a fast release of Twisted with the relevant version skipped in the future, rather than exposing it as a do-it-yourself security fix for end users.